src/lightning/pytorch/core/module.py

```diff
@@ -1403,7 +1403,9 @@ def forward(self, x):
         input_sample = self._on_before_batch_transfer(input_sample)
         input_sample = self._apply_batch_transfer_handler(input_sample)
 
-        torch.onnx.export(self, input_sample, str(file_path), **kwargs)
+        if isinstance(file_path, Path):
+            file_path = file_path.as_posix()
```

Comment:
I think we should keep what #19727 did. We try to avoid making assumptions on posix and keep handling paths OS-independent whenever possible.
```suggestion
        file_path = str(file_path) if isinstance(file_path, Path) else file_path
```

Reply:
Hi @awaelchli 
this is fixed in the new commits. Thanks.